pallets/asset-metadata-extender/src/benchmarking.rs

```diff
@@ -34,6 +124,203 @@ mod benchmarks {
+				claimer.clone(),
+				universal_location.clone(),
+				token_uri,
+			)
+			.unwrap();
+		};
+
+		let new_token_uri: TokenUriOf<T> = vec![2u8; t as usize].try_into().unwrap();
+
+		#[block]
+		{
+			AssetMetadataExtenderPrecompile::<T>::update(
+				&mut handle,
+				universal_location.clone().to_vec().into(),
+				new_token_uri.clone().to_vec().into(),
+			)
+			.unwrap();
+		};
+
+		assert_eq!(
+			AssetMetadataExtender::<T>::token_uris_by_claimer_and_location(
+				claimer,
+				universal_location
+			),
+			Some(new_token_uri)
+		);
+	}
+
+	#[benchmark]
+	fn precompile_balance_of(u: Linear<0, { <T as Config>::MaxUniversalLocationLength::get() }>) {
+		let mut handle = MockHandle::new();
+		let universal_location: UniversalLocationOf<T> = vec![1u8; u as usize].try_into().unwrap();
+
+		#[block]
+		{
+			AssetMetadataExtenderPrecompile::<T>::balance_of(
+				&mut handle,
+				universal_location.clone().to_vec().into(),
+			)
+			.unwrap();
+		};
+	}
+
+	#[benchmark]
+	fn precompile_claimer_by_index(
+		u: Linear<0, { <T as Config>::MaxUniversalLocationLength::get() }>,
+	) {
+		let mut handle = MockHandle::new();
+
+		let caller: T::AccountId = whitelisted_caller();
+		let claimer = caller.clone();
+		let universal_location: UniversalLocationOf<T> = vec![1u8; u as usize].try_into().unwrap();
+		let token_uri: TokenUriOf<T> = vec![1u8; 100usize].try_into().unwrap();
+
+		{
+			AssetMetadataExtender::<T>::create_token_uri_extension(
+				claimer.clone(),
+				universal_location.clone(),
+				token_uri,
+			)
+			.unwrap();
+		};
+
+		#[block]
+		{
+			AssetMetadataExtenderPrecompile::<T>::claimer_by_index(
+				&mut handle,
+				universal_location.clone().to_vec().into(),
+				0u32,
+			)
+			.unwrap();
+		};
+	}
+
+	#[benchmark]
+	fn precompile_extension_by_index(
+		u: Linear<0, { <T as Config>::MaxUniversalLocationLength::get() }>,
+	) {
+		let mut handle = MockHandle::new();
+
+		let caller: T::AccountId = whitelisted_caller();
+		let claimer = caller.clone();
+		let universal_location: UniversalLocationOf<T> = vec![1u8; u as usize].try_into().unwrap();
+		let token_uri: TokenUriOf<T> = vec![1u8; 100usize].try_into().unwrap();
+
+		{
+			AssetMetadataExtender::<T>::create_token_uri_extension(
+				claimer.clone(),
+				universal_location.clone(),
+				token_uri,
+			)
+			.unwrap();
+		};
+
+		#[block]
+		{
+			AssetMetadataExtenderPrecompile::<T>::extension_by_index(
+				&mut handle,
+				universal_location.clone().to_vec().into(),
+				0u32,
+			)
+			.unwrap();
+		};
+	}
+
+	#[benchmark]
+	fn precompile_extension_by_location_and_claimer(
+		u: Linear<0, { <T as Config>::MaxUniversalLocationLength::get() }>,
+	) {
+		let mut handle = MockHandle::new();
+
+		let caller: T::AccountId = whitelisted_caller();
+		let claimer = caller.clone();
+		let universal_location: UniversalLocationOf<T> = vec![1u8; u as usize].try_into().unwrap();
+		let token_uri: TokenUriOf<T> = vec![1u8; 100usize].try_into().unwrap();
+
+		{
```

Comment:
why this block ?

Reply:
if I don't create the extension ahead, `create_token_uri_extension` will fail: https://github.com/freeverseio/laos/blob/a6774a2ca5526c01573864ffd9baf8f5b9a7a617/pallets/asset-metadata-extender/src/precompiles/asset_metadata_extender/mod.rs#L221-L222 and it couldn't be benchmarked